content/en/docs/concepts/containers/container-lifecycle-hooks.md

```diff
@@ -64,11 +64,10 @@ When a Container lifecycle management hook is called,
 the Kubernetes management system executes the handler according to the hook action,
 `httpGet` , `tcpSocket` and `sleep` are executed by the kubelet process, and `exec` is executed in the container.
 
-Hook handler calls are synchronous within the context of the Pod containing the Container.
-This means that for a `PostStart` hook,
-the Container ENTRYPOINT and hook fire asynchronously.
```

Comment:
The word "asynchronously" is wrong here.
If you start a pod and set the `PostStart` to sleep for 180 seconds, it finishes its execution before the ENTRYPOINT begins. This is not the same for the `PreStop` hook due to the TERM signal. Try the below examples and let me know what you think.

**post-start.yaml**
```yaml
apiVersion: v1
kind: Pod
metadata:
  name: post-start
spec:
  containers:
  - name: alpine-container
    image: alpine:latest
    lifecycle:
      postStart:
        exec:
          command: ["/bin/sh", "-c", "echo 'Container has started' > /tmp/poststart.log && sleep 180"]
    command: ["/bin/sh", "-c", "while true; do sleep 30; done"]
```

**pre-stop.yaml**
```yaml
apiVersion: v1
kind: Pod
metadata:
  name: pre-stop
spec:
  containers:
  - name: alpine-container
    image: alpine:latest
    lifecycle:
      preStop:
        exec:
          command: ["/bin/sh", "-c", "echo 'Container is stopping' > /tmp/prestop.log && sleep 180"]
    command: ["/bin/sh", "-c", "while true; do sleep 30; done"]
```

Reply:
ok makes sense